edx-platform/nau-basic/lms/templates/footer.html

```diff
@@ -105,3 +113,91 @@
     <link rel="stylesheet" type="text/css" href="${url}"></link>
   % endfor
 % endif
+
+%if modal_force_display:
+## Pop-up tos
+<div class="modal-container">
+  <div class="modal-content">
+    % if modal_tos_text:
+    ${modal_tos_text | n}
+    <script type="text/javascript">
+    
+    </script>
+
+    % endif
+    <button class="button-modal btn-primary">Accept</button>
+  </div>
+</div>
+
+  %if user.is_authenticated:
+  <% value_preference = get_user_preference(user, name_preference_tos) %>
+
+  <script type="text/javascript">
+
+    const newTosModal = (function () {
+      const valuePreference = '${value_preference}';
+      const urlPreferenceTos = '/api/user/v1/preferences/${user}/${name_preference_tos}';
+      const namePreferenceTos ='${name_preference_tos}';
+      const dateJoined ='${user.date_joined}';
+      const removeWord = namePreferenceTos.slice(-10).replaceAll('_', '-') 
+      const getDateJoined = new Date(dateJoined);
+      const getDateModalTos = new Date(removeWord);
+      
+      function userJoinedAfterTosDate() {
```

Comment:
I do not understand why change what you had for this function. In general what is inside the function is not necessary, you could have done `const userJoinedAfterTosDate = getDateJoined.getTime() >= getDateModalTos.getTime();` and with that was what you needed. In the long run `valueCompare` will return the same value from `valueCompareDates`